### PR TITLE
[bcl] Build some assemblies with -warnaserror

### PR DESCRIPTION
**Pull request by akoeplinger**
This ensures that we keep the build warning free.

We'll only opt-in a few assemblies that already have zero warnings for now.
We also only apply this when using csc as mcs might raise different warnings.

@marek-safar thoughts?

### CODE REVIEW

**Comment by marek-safar** [comment]
I am ok with it if it's opt-in option but we need to ensure we pass on mcs too as mcs reports more warnings than csc

**Comment by alexrp** [comment]
Feel free to add Mono.Profiler.Log here. It should be warning-free already.

File: mcs/build/library.make [hunk 1]
```diff
@@ -44,6 +44,12 @@ RESOURCE_STRINGS_FILES += $(RESOURCE_STRINGS:%=--resourcestrings:%)
 endif
 endif
 
+ifdef LIBRARY_WARN_AS_ERROR
```
Comment (vargaz) [inline mcs/build/library.make]: This shouldn't be done in the basic profile, which uses system mcs/csc which we don't control.

**Comment by akoeplinger** [comment]
@monojenkins build

**Comment by alexrp** [comment]
What's the status on this PR?

**Comment by marek-safar** [comment]
build

**Comment by akoeplinger** [comment]
@alexrp I got sidetracked with other stuff, rebased now.

**Comment by luhenry** [comment]
Shouldn't we invert `LIBRARY_WARN_AS_ERROR` to `DISABLE_LIBRARY_WARN_AS_ERROR` and define the latter only on assemblies that we don't want to (or can't) fix right now? I assume there is only a handful of assemblies that do have warning and that falls into this category, and it would allow us to cover more ground by default.

**Comment by vargaz** [comment]
We have lots of assemblies with warnings.

**Comment by akoeplinger** [comment]
Correct, there are still a bunch of assemblies with warnings and I wanted to make it opt-in at first so we don't inadvertently break something.

**Pull request merged**